agent/tool-scripts/postprocess/unittests

```diff
@@ -70,30 +71,18 @@ function testit {
     return ${sts}
 }
 
-# Remove any file that is in the sample directory that
-# is still the same in the test directory, removing any
-# empty directories first.
+# Remove any file from the test directory that is the same as the
+# corresponding file in the sample directory (or is a symlink), and
+# then recursively remove any empty directories.
 function remove_sample_data {
-    local i
-    # First remove symlinks
-    for i in $(cd samples/$1; find . -type l); do
-        rm -rf ${_testroot}/$1/$i
-    done
-    # Remove all files that have not changed from the
-    # original sample
-    for i in $(cd samples/$1; find . -type f); do
-        diff -cq samples/$1/$i ${_testroot}/$1/$i > /dev/null 2>&1 && rm -f ${_testroot}/$1/$i
-    done
-    let dir_cnt=-1
-    while [[ $dir_cnt -gt 0 || $dir_cnt -lt 0 ]]; do
-        let dir_cnt=0
-        # In the loop, exit condition set should we
-        # find no empty directories.
-        for i in $(find ${_testroot}/$1 -type d -empty); do
-            let dir_cnt=$dir_cnt+1
-            rmdir $i
-        done
-    done
+    local cur_dir=${PWD}
+    pushd ${_testroot}/$1 >/dev/null
+    find . \
+      \( -type l \
+      -or -type f -and -exec diff -q ${cur_dir}/samples/$1/{} {} \; \
+      -or -type d -and -empty \) \
+      -delete >/dev/null 2>&1
+    popd >/dev/null
```

Comment:
Can't we execute the `find` command with `${_testroot}/$1` instead of `.` and save the `pushd`/`popd`?

Reply:
No, we can't -- we need the `find` command to generate local/relative paths which we can append to our current working directory.